plugins/node/opentelemetry-instrumentation-mongodb/package.json

```diff
@@ -58,7 +58,7 @@
     "@types/node": "16.11.21",
     "gts": "3.1.0",
     "mocha": "7.2.0",
-    "mongodb": "3.6.11",
+    "mongodb": "4.6.0",
```

Comment:
Having mongodb in dev deps only will result in missing types for consumers.
Adding a specific mongodb version as dependency seems not good.

Reply:
I agree. thats why this is a draft. i'm trying to figure out other options. Unfortunately, I think we're going to have to stop trying to type check modules we're patching. causing too many dependency troubles.

Reply:
For the patching part it's quite simple as the instrumentation base classes don't need/use the types at all.
The problem are the hooks offered by some instrumentations as they are APIs to end users. We could use `unknown`/`any` there and just document what is provided and user as to cast. This would even cover differences between versions.